packages/lib-react-components/src/FavoritesIconButton/FavoritesIconButtonContainer.js

```diff
@@ -0,0 +1,127 @@
+import { shape, string } from 'prop-types'
+import useSWRMutation from 'swr/mutation'
+
+import {
+  usePanoptesAuthToken,
+  useUserCollections
+} from '../hooks'
+import {
+  addSubjectsToCollection,
+  removeSubjectsFromCollection
+} from '../helpers/collections'
+import FavoritesIconButton from './FavoritesIconButton'
+
+function FavoritesIconButtonContainer({
+  project,
+  subject,
+  user,
+  ...props
+}) {
+  const token = usePanoptesAuthToken()
+
+  const query = {
+    favorite: true,
+    project_ids: [project?.id],
+    owner: user?.login
+  }
+
+  const { trigger: addToFavorites } = useSWRMutation({ query, token }, addSubjectsToCollection)
+  const { trigger: removeFromFavorites } = useSWRMutation({ query, token }, removeSubjectsFromCollection)
```

Comment:
I don't like including a `token` here, but I think for `swr` mutations to work as intended then the first param should be the key per `useUserCollections`?